Flaky test TestDelayedCompaction/delayed_compaction_enabled on windows

While bumping dependencies, sometimes there was a test on windows which failed. A retry solve the problem:

```
=== NAME  TestDelayedCompaction
    compact_test.go:2045: 
        	Error Trace:	D:/a/prometheus/prometheus/tsdb/compact_test.go:1950
        	            				D:/a/prometheus/prometheus/tsdb/compact_test.go:2045
        	Error:      	"2.0426994s" is not less than "2s"
        	Test:       	TestDelayedCompaction
--- FAIL: TestDelayedCompaction (3.16s)
    --- PASS: TestDelayedCompaction/delayed_compaction_not_enabled (0.41s)
    --- FAIL: TestDelayedCompaction/delayed_compaction_enabled (3.16s)
panic: test executed panic(nil) or runtime.Goexit

goroutine 854 [running]:
testing.tRunner.func1.2({0x136d6a0, 0x2141a00})
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1734 +0x219
testing.tRunner.func1()
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1737 +0x359
runtime.Goexit()
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/runtime/panic.go:636 +0x5e
testing.(*common).FailNow(0xc000106fc0)
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1041 +0x4a
github.com/stretchr/testify/require.Less({0x182ea90, 0xc000106fc0}, {0x1483660, 0xc0004a8020}, {0x1483660, 0xc0004a8028}, {0x0, 0x0, 0x0})
	C:/Users/runneradmin/go/pkg/mod/github.com/stretchr/testify@v1.10.0/require/require.go:1188 +0xff
github.com/prometheus/prometheus/tsdb.TestDelayedCompaction.func1(0xc000186120)
	D:/a/prometheus/prometheus/tsdb/compact_test.go:1950 +0x173
github.com/prometheus/prometheus/tsdb.TestDelayedCompaction.func3(0xc000107a40)
	D:/a/prometheus/prometheus/tsdb/compact_test.go:2045 +0x4cd
testing.tRunner(0xc000107a40, 0xc001718a20)
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1792 +0xcb
created by testing.(*T).Run in goroutine 852
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1851 +0x3f6
FAIL	github.com/prometheus/prometheus/tsdb	13.629s
```

OR

```
--- FAIL: TestDelayedCompaction (3.18s)
    --- PASS: TestDelayedCompaction/delayed_compaction_not_enabled (0.29s)
    --- FAIL: TestDelayedCompaction/delayed_compaction_enabled (3.17s)
panic: test executed panic(nil) or runtime.Goexit

goroutine 936 [running]:
testing.tRunner.func1.2({0x13ed6a0, 0x21c1a00})
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1734 +0x219
testing.tRunner.func1()
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1737 +0x359
runtime.Goexit()
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/runtime/panic.go:636 +0x5e
testing.(*common).FailNow(0xc0000036c0)
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1041 +0x4a
github.com/stretchr/testify/require.Less({0x18aea90, 0xc0000036c0}, {0x1503660, 0xc0004cc078}, {0x1503660, 0xc0004cc080}, {0x0, 0x0, 0x0})
	C:/Users/runneradmin/go/pkg/mod/github.com/stretchr/testify@v1.10.0/require/require.go:1188 +0xff
github.com/prometheus/prometheus/tsdb.TestDelayedCompaction.func1(0xc0004e5200)
	D:/a/prometheus/prometheus/tsdb/compact_test.go:1950 +0x173
github.com/prometheus/prometheus/tsdb.TestDelayedCompaction.func3(0xc000003a40)
	D:/a/prometheus/prometheus/tsdb/compact_test.go:2045 +0x4cd
testing.tRunner(0xc000003a40, 0xc001d8fad0)
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1792 +0xcb
created by testing.(*T).Run in goroutine 934
	C:/hostedtoolcache/windows/go/1.24.2/x64/src/testing/testing.go:1851 +0x3f6
FAIL	github.com/prometheus/prometheus/tsdb	13.649s
```

Refs:

* https://github.com/prometheus/prometheus/actions/runs/14521987145/job/40744880390?pr=16373
* https://github.com/prometheus/prometheus/actions/runs/14518952318/job/40734875315?pr=16372

--- a/tsdb/compact_test.go
+++ b/tsdb/compact_test.go
@@ -1988,7 +1988,7 @@ func TestDelayedCompaction(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
-			if c.compactionDelay > 0 && runtime.GOOS == "windows" {
+			if runtime.GOOS == "windows" {
 				t.Skip("Time imprecision on windows makes the test flaky, see https://github.com/prometheus/prometheus/issues/16450")
 			}
 			t.Parallel()

